File: go/space-age/space_age.go
package space

type Planet string

func Age(second float64, planet Planet) (result float64){
  s := map[Planet]float64{
    "Earth": 31557600,
    "Mercury": 7600543.81992,
    "Venus": 19414149.052176,
    "Mars": 59354032.69008,
    "Jupiter": 374355659.124,
    "Saturn": 929292362.8848,
    "Uranus": 2651370019.3296,
    "Neptune": 5200418560.032,
  }
  second_per_year := s[planet]
  result = second / second_per_year
  return result
}

File: dart/space-age/lib/space_age.dart
class SpaceAge {
  final ages = <String, double>{
    'Mercury': 0.2408467,
    'Venus': 0.61519726,
    'Earth': 1,
    'Mars': 1.8808158,
    'Jupiter': 11.862615,
    'Saturn': 29.447498,
    'Uranus': 84.016846,
    'Neptune': 164.79132
  };

  double age({String planet, int seconds}) {
    final int secondsInYear = 31557600;
    var earthYear = ages[planet];
    return double.parse((seconds / (secondsInYear * earthYear)).toStringAsFixed(2));
  }
}
